### ISSUE TITLE
More than 8 columns display in an unexpected order

### ISSUE THREAD

**Issue by erinknight242** [GriddleGriddle/Griddle#804]
## Griddle version
v.1.11.2

## Expected Behavior
Columns will display in the order listed in the data (like it does for <=8 columns)

## Actual Behavior
Columns appear in an unexpected order (7, 8, 9, 1, 2, 3, 4, 5, 6)

## Steps to reproduce
Follow the Getting Started example, but with data with >8 columns:
```javascript
<Griddle
  data={[
    { col1: '1', col2: '2', col3: '3', col4: '4', col5: '5',
      col6: '6', col7: '7', col8: '8', col9: '9'}
  ]}
  plugins={[plugins.LocalPlugin]}
/>
```

**Comment by dahlbyk** [comment]
Interesting. I would expect this to have been fixed by https://github.com/GriddleGriddle/Griddle/pull/658/commits/3a742520cdb67bb40b97fd55b70dc5ad46404b62 in https://github.com/GriddleGriddle/Griddle/pull/658.

Can you please try the new story added in #805? I'm not able to reproduce this.
![image](https://user-images.githubusercontent.com/133987/38273613-067f36c6-3752-11e8-8f9a-944ef349a5c4.png)

**Comment by erinknight242** [comment]
Agree; interesting. The storybook columns look correct for me as well; not sure why v1.11.2 installed in my app isn't working properly. I'll try to repro elsewhere.

**Comment by erinknight242** [comment]
Ok; I was able to repro it. It seems to be related to my (rather undesirable) column names, which are formatted like a mm-dd date.

```javascript
data={[
  { '00-00': '0', '01-01': '1', '02-02': '2', '03-03': '3', '04-04': '4',
    '05-05': '5', '06-06': '6', '07-07': '7', '08-08': '8', '09-09': '9', '10-10': '10' }
]}
```

**Comment by dahlbyk** [comment]
Confirmed. No idea why. I just pushed a failing test to #805, if you care to look into what the problem may be in [`dataSelectors.js`](https://github.com/GriddleGriddle/Griddle/blob/8facf9a5a2e0024593b6205077a0e68fc8455b85/src/selectors/dataSelectors.js).
